src/speed-index.js

```diff
@@ -33,14 +33,44 @@ function calculateFrameProgress(current, initial, target) {
 	return progress;
 }
 
-function calculateVisualProgress(frames) {
+function calculateProgressBetweenFrames(frames, lowerBound, upperBound, threshold, getProgress, setProgress) {
+	if (threshold === 0) {
+		frames.forEach(frame => setProgress(frame, getProgress(frame)));
+		return;
+	}
+
+	const lowerFrame = frames[lowerBound];
+	const upperFrame = frames[upperBound];
+
+	const lowerProgress = getProgress(lowerFrame);
+	const upperProgress = getProgress(upperFrame);
+
+	setProgress(lowerFrame, lowerProgress);
+	setProgress(upperFrame, upperProgress);
+
+	if (Math.abs(lowerProgress - upperProgress) < threshold) {
```

Comment:
as discussed earlier i'm thinking we might want to threshold on time as well.

```js
const FastMode_FrameProgressDelta = 5;  // units: percent
const FastMode_TimeDelta = 200;   // units: ms
```

wdyt? 
this way we can handle a 5% change over 5 seconds (probably worth inspecting) vs a 5% change over 100ms (not worth it)